stlab/concurrency/future.hpp

```diff
@@ -296,21 +297,24 @@ struct shared_base<T, enable_if_copyable<T>> : std::enable_shared_from_this<shar
     template <typename F, typename... Args>
     void set_value(F& f, Args&&... args);
 
-    auto get_try() -> boost::optional<T> {
+    auto get_try() -> boost::optional<const T&> {
+        boost::optional<const T&> result{ boost::none };
         bool ready = false;
         {
             std::unique_lock<std::mutex> lock(_mutex);
             ready = _ready;
         }
         if (ready) {
             if (_error) std::rethrow_exception(_error.get());
-            return _result;
+            result = _result.get();
         }
-        return boost::none;
+        return result;
     }
 
     auto get_try_r(bool unique) -> boost::optional<T> {
-        if (!unique) return get_try();
+        // TODO Fp I think that is not needed, because an r-value
```

Comment:
The unique test is needed - I can have multiple items with the shared same state, even if they are rvalues. So this is "if this is an rvalue and the state is not shared" -
Example:
```cpp
future<int> x = f();
auto y = x; // copy
move(y).get_try(); // y and x will share state despite rvalue, can't optimize!
```